lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb

```diff
@@ -0,0 +1,214 @@
+          end
+
+          # Creates the Info.plist file which sets public framework attributes
+          #
+          # @return [void]
+          #
+          def create_info_plist_file
+            path = target.info_plist_path
+            UI.message "- Generating Info.plist file at #{UI.path(path)}" do
+              generator = Generator::InfoPlistFile.new(target)
+              generator.save_as(path)
+              add_file_to_support_group(path)
+
+              native_target.build_configurations.each do |c|
+                relative_path = path.relative_path_from(sandbox.root)
+                c.build_settings['INFOPLIST_FILE'] = relative_path.to_s
+              end
+            end
+          end
+
+          # Creates the module map file which ensures that the umbrella header is
+          # recognized with a customized path
+          #
+          # @yield_param [Generator::ModuleMap]
+          #              yielded once to configure the private headers
+          #
+          # @return [void]
+          #
+          def create_module_map
+            path = target.module_map_path
+            UI.message "- Generating module map file at #{UI.path(path)}" do
+              generator = Generator::ModuleMap.new(target)
+              yield generator if block_given?
+              generator.save_as(path)
+              add_file_to_support_group(path)
+
+              native_target.build_configurations.each do |c|
+                relative_path = path.relative_path_from(sandbox.root)
+                c.build_settings['MODULEMAP_FILE'] = relative_path.to_s
+              end
+            end
+          end
+
+          # Generates a header which ensures that all header files are exported
+          # in the module map
+          #
+          # @yield_param [Generator::UmbrellaHeader]
+          #              yielded once to configure the imports
+          #
+          def create_umbrella_header
+            path = target.umbrella_header_path
+            UI.message "- Generating umbrella header at #{UI.path(path)}" do
+              generator = Generator::UmbrellaHeader.new(target)
+              yield generator if block_given?
+              generator.save_as(path)
+
+              # Add the file to the support group and the native target,
+              # so it will been added to the header build phase
+              file_ref = add_file_to_support_group(path)
+              native_target.add_file_references([file_ref])
+
+              # Make the umbrella header public
+              build_file = native_target.headers_build_phase.build_file(file_ref)
+              build_file.settings ||= {}
+              build_file.settings['ATTRIBUTES'] = ['Public']
+            end
+          end
+
+          # Generates a dummy source file for each target so libraries that contain
+          # only categories build.
+          #
+          # @return [void]
+          #
+          def create_dummy_source
```

Comment:
@endocrimes  it's a long time past,  do you still remember  the reason for create the dummy files .